components/ui/AppMegaDropdownMenu.vue

```diff
@@ -75,90 +74,121 @@ export default class TheMegaDropdownMenu extends Vue {
   }
 
   filterText = ''
-  get filterWords (): string[] {
+
+  filterWords (): string[] {
     return this.filterText.trim().toLowerCase().split(' ')
   }
 
-  highlightParts (text: string) {
-    if (this.filterText.trim() === '' || text.trim() === '') {
-      return [{ text, hightlight: false }]
+  filterTextIsEmpty (): boolean {
+    return this.filterText.trim() === ''
+  }
+
+  splitTextInHighlightParts (text: string) : HighlightTextState[] {
+    const textIsEmpty = text.trim() === ''
+    if (this.filterTextIsEmpty() || textIsEmpty) {
+      return [{ text, highlight: false }]
     }
 
-    const highlightStateByChar = this._highlightStateByChar(text)
+    const charIsHighlightArray = this._splitTextInHighlightChars(text, this.filterWords())
 
-    const output = this._concatByHighlightState(highlightStateByChar)
+    const textHighlightParts = this._joinCharsByHighlightState(charIsHighlightArray)
 
-    return output
+    return textHighlightParts
   }
 
-  _highlightStateByChar (text: string) : HighlightTextState[] {
-    const highlightState = Array.from(text).map<HighlightTextState>((letter:string) => { return { text: letter, highlight: false } })
-
+  // Splits the text in characters and sets a "highlight" property indicating if the character should be highlighted or not
+  _splitTextInHighlightChars (text: string, filterWords: string[]) : HighlightTextState[] {
+    const charArray = Array.from(text)
+    // Assign a highlight flag to each character
+    const highlightStates = charArray.map<HighlightTextState>(letter => ({ text: letter, highlight: false }))
```

Comment:
Could you use other naming for `text`? maybe `menuElementChar`?

`highlight` is a boolean, maybe `isHighligthed` helps reading the code